kolibri/auth/test/test_morango_integration.py

```diff
@@ -48,3 +61,173 @@ def test_deserializing_field(self):
             self.controller.deserialize_from_store()
         except AttributeError as e:
             self.fail(e.message)
+
+
+@unittest.skipIf(sys.platform.startswith("win"), "can't run on Windows")
+@unittest.skip("Test will be run manually for now because it will greatly increase testing times.")
+class EcosystemTestCase(TestCase):
+
+    def setUp(self):
+        pass
+
+    def _data(self, *args, **kwargs):
+        return kwargs
+
+    def _create_objects(self, server):
+        fac = Facility.objects.using(server.db_alias).first()
+        admin = FacilityUser(username=uuid.uuid4().hex[:30], facility=fac)
+        admin.save(using=server.db_alias)
+        learner = FacilityUser(username=uuid.uuid4().hex[:30], facility=fac)
+        learner.save(using=server.db_alias)
+        class_resp = self.request_server(server, "classroom", data=self._data(parent=fac.id, name=uuid.uuid4().hex))
+        lg_resp = self.request_server(server, "learnergroup", data=self._data(parent=class_resp.json()['id'], name=uuid.uuid4().hex))
+        self.request_server(server, "membership", data=self._data(user=learner.id, collection=class_resp.json()['id']))
+        self.request_server(server, "membership", data=self._data(user=learner.id, collection=lg_resp.json()['id']))
+        self.request_server(server, "role", data=self._data(collection=fac.id, user=admin.id, kind="admin"))
+
+    def request_server(self, server, endpoint, method="POST", lookup=None, data={}, params={}):
+        """
+
+        :param server: kolibri instance we are querying
+        :param endpoint: constant representing which kolibri endpoint we are querying
+        :param method: HTTP verb/method for request
+        :param lookup: the pk value for the specific object we are querying
+        :param data: dict that will be form-encoded in request
+        :param params: dict to be sent as part of URL's query string
+        :return: ``Response`` object from request
+        """
+
+        # build up url and send request
+        if lookup:
+            lookup = lookup + '/'
+        url = urljoin(urljoin(server.base_url, "api/" + endpoint + "/"), lookup)
+        auth = ("superuser", "password")
+        resp = requests.request(method, url, json=data, params=params, auth=auth)
+        resp.raise_for_status()
+        return resp
+
+    def assertServerQuerysetEqual(self, s1, s2):
```

Comment:
Helper test function which compares all the syncable models between two servers to ensure they are equal.